### ISSUE TITLE
Still memory leak 

### ISSUE THREAD

**Issue by Duartteno** [SkriptLang/Skript#1763]
### Description

Local variables still causes memory leaks

### Steps to Reproduce
```
command /test:
  trigger:
    set {_i} to 1
    set {test} to true
    while {test} is true:
      add 1 to {_i}
      loop 1000 times:
        set {_test::%{_i}%::%loop-number-2%} to random number between 1 and 999999999
        send "Ram: %total ram - free ram% MB / %total ram% MB"
      wait 1 tick

command /stoptest:
  trigger:
    set {test} to false
```

### Server Information
* **Server version/platform:**       1.12.2
* **Skript version:**                2.3-beta4

**Comment by Duartteno** [comment]
ok i found it...
```
set {_test::%loop-number-2%} to random number between 1 and 999999999 #works
```
leak
```
set {_test::%{_n}%::%loop-number-2%} to random number between 1 and 999999999
```

**Comment by bensku** [comment]
Working as intended. You're not exiting the trigger until you use /stoptest, thus local variables are not removed. The "fix" you found relies on the script overwriting existing variables instead of creating new ones.

**Comment by Duartteno** [comment]
@bensku after type `/stoptest` RAM memory stays at 2Gb and it will never decrease until the restart.

**Comment by bensku** [comment]
JVM does not necessarily free RAM to OS, at least not until version 12 (see https://openjdk.java.net/jeps/346). Unless the expressions you use to get information about RAM have confusing names and refer to heap instead of physical memory, this is not an issue with Skript.

**Comment by Duartteno** [comment]
@bensku 
<img width="853" alt="leak1" src="https://user-images.githubusercontent.com/44345095/50385914-9b9ff000-06dd-11e9-9a27-b7e4b0ba1843.png">
![Uploading leak2.png…]()

**Comment by bensku** [comment]
Alright, now *this* is looking troubling.

**Comment by Duartteno** [comment]
@bensku tested, issue still exists. I have no idea why.

**Comment by bensku** [comment]
Cannot reproduce. I did find a (relatively unlikely) memory leak, but if was fixed by commit 237b3dd.

**Comment by bensku** [comment]
@TheBentoBox Did you find a way to reproduce this?

**Comment by TheBentoBox** [comment]
No, I just labelled it since it has been sitting for months unlabelled and with no activity, but if we're not going to do anything until someone comes back with reproduction steps then we should probably just let that be reported as a new issue and close this one.

**Comment by bensku** [comment]
No one has been able to reproduce this with recent Skript versions, so closing.
